cfme/cloud/instance/__init__.py

```diff
@@ -192,318 +224,594 @@ def find_quadicon(self, *args, **kwargs):
+        self.view.instances.tree.click_path('All Instances')
 
-        # use accordion
+    def resetter(self, *args, **kwargs):
         # If a filter was applied, it will persist through navigation and needs to be cleared
         if sel.is_displayed(search_box.clear_advanced_search):
             logger.debug('Clearing advanced search filter')
             sel.click(search_box.clear_advanced_search)
-        accordion.tree('Instances', 'All Instances')
+
+
+class ProviderAllView(CloudInstanceView):
+    @property
+    def is_displayed(self):
+        return (
+            self.in_cloud_instance and
+            self.title.text == 'Instances under Provider "{}"'
+                               .format(self.context['object'].provider.name)
+        )
 
 
 @navigator.register(Instance, 'AllForProvider')
-class InstanceAllForProvider(CFMENavigateStep):
+class AllForProvider(CFMENavigateStep):
+    VIEW = ProviderAllView
     prerequisite = NavigateToAttribute('appliance.server', 'LoggedIn')
 
-    def am_i_here(self):
-        return match_page(summary='Instances under Provider "{}"'.format(self.obj.provider.name))
-
     def step(self, *args, **kwargs):
         self.prerequisite_view.navigation.select('Compute', 'Clouds', 'Instances')
+        self.view.instances_by_provider.tree.click_path('Instances by Provider',
+                                                        self.obj.provider.name)
 
-        # use accordion
+    def resetter(self, *args, **kwargs):
         # If a filter was applied, it will persist through navigation and needs to be cleared
         if sel.is_displayed(search_box.clear_advanced_search):
             logger.debug('Clearing advanced search filter')
             sel.click(search_box.clear_advanced_search)
-        accordion.tree('Instances by Provider', 'Instances by Provider', self.obj.provider.name)
+
+
+class AddFloatIPView(CloudInstanceView):
+    @View.nested
+    class form(View):  # noqa
+        ip = BootstrapSelect('floating_ip')
+        submit_button = Button('Submit')
+        cancel_button = Button('Cancel')
+
+    @property
+    def is_displayed(self):
+        # Only the instance name is displayed, cannot confirm provider
+        return False
+
+
+@navigator.register(Instance, 'AddFloatingIP')
+class AddFloatingIP(CFMENavigateStep):
+    VIEW = AddFloatIPView
+    prerequisite = NavigateToSibling('Details')
+
+    def step(self):
+        if version.current_version() >= '5.7':
+            try:
+                self.prerequisite_view.configuration.item_select(
+                    'Associate a Floating IP with this Instance')
+            except NoSuchElementException:
+                raise DestinationNotFound('Add Floating IP option not available for instance')
+        else:
+            raise DestinationNotFound('Floating IP assignment not available for appliance version')
+
+
+class AttachVolumeView(CloudInstanceView):
+    @View.nested
```

Comment:
This made more sense to me, because not all widgets in the view are part of the form, and some views have a form and a display (like timelines). Checking out toolbar view structure from your other comment.